doc/source/metadata/Working-with-Metadata.rst

```diff
@@ -0,0 +1,295 @@
+is if organisation-specific details - such as the name of the model ID attribute
+- need to be passed in via the command line.
+Even in these cases, the plugin should take as much responsibility as possible,
+requiring minimal information from the user to inform metadata updates.
+(For example, in the model ID attribute case,
+the user is required to provide the name of the attribute from which to read
+model information, rather than a ``name: value`` pairing
+to be directly applied.)
+This maximises code flexibility and minimises the chances of
+bugs or inconsistencies by clearly recording the expected metadata
+within the code, where it can be covered by automated tests.
+
+Implementing a step in the suite
+--------------------------------
+
+Metadata is almost exclusively dealt with at the code level,
+with plugins taking responsibility for updating the
+appropriate metadata internally.
+However, there are a few limited cases where the code needs information
+to be provided via command line in order to make the correct updates:
+
+Standardisation
+    In the Met Office implementaiton, the “standardise” step at the start of
+    each suite chain has been configured to remove unnecessary attributes
+    from incoming data.
+
+New parameters
+    If a new parameter is to be blended,
+    the name of the model ID attribute needs to be provided via the suite app
+    so that this attribute can be included on the parameter file.
+    If this argument is omitted,
+    the file will not contain source model information and will not be able
+    to be blended.
+
+Spot extracted data
+    This requires a ``title``, which must currently be provided
+    via a command line argument.
+    If not provided, the title will default to ``unknown``.
+
+Modifying existing functions or plugins
+---------------------------------------
+
+When modifying an existing function or plugin it will not usually be necessary
+to change how metadata are treated.
+However, it is worth developers considering the following specific questions:
+
+* Have I significantly changed the amount of post-processing
+  this plugin is doing?
+  If so, does it need to change from a ``BasePlugin``
+  to a ``PostProcessingPlugin`` or vice versa?
+* Have I changed what this plugin is doing,
+  i.e. from producing coefficients or generating a correction to applying them?
+  Does it now need to be a ``PostProcessingPlugin``
+  where previously it was a general object?
+* Is this plugin as a whole taking the right level of responsibility
+  for the changes it is making?
+  Are there any public methods that take only partial responsibility,
+  and so should be private?
+* Should this function be a plugin (e.g. feels_like_temperature)?
+
+Some of these are 'nice-to-have' questions, which should be considered
+if refactoring a piece of code more widely
+(as opposed to one-line changes or small bug fixes),
+to help guide the new design.
+
+Using the metadata interpreter
```

Comment:
I think this section needs to cover the possibility that the metadata interpreter may need to be modified / updated to accommodate new metadata that is required. It may only be a statement to acknowledge that this is a possibility, but without such a line it seems as if any non-compliance is unacceptable, whereas this tool is really only a guide based on the current standard.